Source/ACE.Database/ShardDatabase.cs

```diff
@@ -173,9 +174,15 @@ public Task RemoveAllFriends(uint characterId)
             throw new NotImplementedException();
         }
 
-        public void DeleteOrRestore(ulong unixTime, uint id)
+        public async Task<bool> DeleteOrRestore(ulong unixTime, uint aceObjectId)
         {
-            throw new NotImplementedException();
+            AceCharacter aceCharacter = new AceCharacter(aceObjectId);
+            LoadIntoObject(aceCharacter);
+            aceCharacter.DeleteTime = unixTime;
+
+            // aceCharacter.Deleted = true;  This is just a reminder - the DB will set this 1 hour after deletion.
```

Comment:
As a protection we should put `aceCharacter.Deleted = false` which also makes this work for Restore.